lib/scala/syntax/definition/src/main/scala/org/enso/syntax/text/spec/DocParserDef.scala

```diff
@@ -252,7 +252,7 @@ case class DocParserDef() extends Parser[Doc] {
       logger.trace {
         var listOfFormattedAST: List[Elem] = Nil
         while (
-          result.stack.head != Elem.Formatter(typ) && result.stack.nonEmpty
+          result.stack.nonEmpty && result.stack.head != Elem.Formatter(typ)
```

Comment:
This check was originally in wrong order - if `result.stack` was empty the latter would be false, but before it's checked we'd call `result.stack.head` which would throw an exception (due to an empty stack).

For some reason the changes in this PR have started triggering this. It is a bit curious why, but I think we should just have it changed to correctly finish the loop if the stack is empty.